### PR TITLE
Cart estimate shipping

### PR DESCRIPTION
**Pull request by jnurse**
This PR replaces the hardcoded shipping rate with correct values, and implements the estimate shipping modal submission

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1110
 **Linked PRs**: n/a

## Changes
- Refactored estimate shipping API call to be re-used on cart & checkout shipping pages
- Fetched shipping method & process checkout data when cart data is received
- Wrapped estimate shipping form in redux-form so we can access selected values
- Added action to submit estimate shipping form 

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Add an item to your cart
- click "calculate" on the cart page
- enter values in the estimate shipping form and submit
- a network request should go out with the values you entered (note all shipping is the same regardless of location right now, so shipping rate will not change)
- proceed to the checkout shipping page
- enter values for country, region and postal code
- confirm that a network request goes out with the new values entered (again shipping rate will not change)

### CODE REVIEW

**Comment by fractaltheory** [comment]
I think this change would benefit from region filtering, implemented in another PR: https://github.com/mobify/progressive-web-scaffold/pull/337

File: web/app/containers/checkout-shipping/actions.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ export const process = ({payload: {$, $response}}) => {
     return receiveData(checkoutShippingParser($, $response))
 }
 
+
+export const fetchShippingMethods = () => fetchShippingMethodsEstimate('shippingForm')
```
Comment (fractaltheory) [inline web/app/containers/checkout-shipping/actions.js]: We could use the `SHIPPING_FORM_NAME` constant string here, right?

Comment (jnurse) [inline web/app/containers/checkout-shipping/actions.js]: Yup, good catch! 😄

File: web/app/store/checkout/shipping/actions.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            address.country_id = formValues.country_id
+            address.region_id = formValues.region_id
+            address.postcode = formValues.postcode
+            if (formValues.region) {
+                address.region = formValues.region
+                // Remove the region_id in case we have an old value
+                delete address.region_id
+            }
+        }
+        const getEstimateURL = `https://www.merlinspotions.com/rest/default/V1/${isLoggedIn ? 'carts/mine' : `guest-carts/${entityID}`}/estimate-shipping-methods`
```
Comment (fractaltheory) [inline web/app/store/checkout/shipping/actions.js]: I think we've been trying to use paths that don't include the hostname, to allow for compatibility with staging environments.

File: web/app/store/checkout/shipping/actions.js [hunk 3]
```diff
@@ -2,30 +2,37 @@
...
 
 
+
 export const fetchShippingMethodsEstimate = (formKey) => {
     return (dispatch, getState) => {
         const currentState = getState()
         const isLoggedIn = getIsLoggedIn(currentState)
         const formValues = getFormValues(formKey)(currentState)
         const entityID = getCustomerEntityID(currentState)
+        const registeredFieldNames = getFormRegisteredFields(formKey)(currentState).map(({name}) => name)
```
Comment (fractaltheory) [inline web/app/store/checkout/shipping/actions.js]: `getFormRegisteredFields(formKey)(currentState)` can returned `undefined` which will throw an error when `map` is called.

**Comment by fractaltheory** [comment]
Seems to work 👍

**Pull request merged**